### PR TITLE
Only pass the protocol in use in to establish a WS connection

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16114)

### CODE REVIEW

**Comment by nox** [comment]
This removes the dependency on websocket from the script crate, which is a good thing to do to try to switch to tungstenite.

File: components/net/websocket_loader.rs [hunk 1]
```diff
@@ -77,9 +78,12 @@ fn establish_a_websocket_connection(resource_url: &ServoUrl,
 
     // Step 13 and transitive subset of step 14.
     // See step 6 of http://tools.ietf.org/html/rfc6455#section-4.1.
-    if let Some(protocol_name) = unwrap_websocket_protocol(response.protocol()) {
-            if !protocols.is_empty() && !protocols.iter().any(|p| (&**p).eq_ignore_ascii_case(protocol_name)) {
-                return Err(WebSocketError::ProtocolError("Protocol in Use not in client-supplied protocol list"));
+    let protocol_in_use = response.protocol().and_then(|header| {
+        header.first().cloned()
```
Comment (KiChjang) [inline components/net/websocket_loader.rs]: Should add a reference to https://github.com/whatwg/fetch/issues/515.

**Review by KiChjang — DISMISSED** [review]

**Comment by KiChjang** [comment]
r=me once you fix that small nit.

File: Cargo.lock [hunk 2]
```diff
@@ -2293,7 +2293,6 @@ dependencies = [
  "selectors 0.18.0",
  "serde 0.9.11 (registry+https://github.com/rust-lang/crates.io-index)",
  "serde_json 0.9.9 (registry+https://github.com/rust-lang/crates.io-index)",
- "servo-websocket 0.18.0 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (Zl6h901) [inline Cargo.lock:2296]: A

Comment (Zl6h901) [reply Cargo.lock:2296]: A

**Review by Zl6h901 — COMMENTED** [review]
C

**Pull request merged**